### PR TITLE
Setup test suite

### PR DESCRIPTION
**Pull request by Awarua-**

### CODE REVIEW

File: test/client.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        done: done
+    };
+};
+
+Client.prototype.shutdown = function(cb) {
+    io.emit('message', {
+        content: '/shutdown'
+    });
+    stopCallback = cb;
+}
```
Comment (mrkno) [inline test/client.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/theprojectshop/Kassy/file/2531448133/issues/source?bid=3282987&fileBranchId=3283883#l36)

Comment (mrkno) [inline test/client.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/theprojectshop/Kassy/file/2531448133/issues/source?bid=3282987&fileBranchId=3283883#l36)

File: test/acceptance/test_core_modules.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+var assert = require('chai').assert,
+    c = require('../client.js');
+
+describe('Test core modules', function() {
+    this.timeout(5000);
+    before(function(done) {
+        client = new c(done);
```
Comment (mrkno) [inline test/acceptance/test_core_modules.js]: Codacy Issue found: [Variable 'client' implicitly declared.](https://www.codacy.com/app/theprojectshop/Kassy/file/2531448132/issues/source?bid=3282987&fileBranchId=3283883#l7)

Comment (mrkno) [inline test/acceptance/test_core_modules.js]: Codacy Issue found: [A constructor name should not start with a lowercase letter.](https://www.codacy.com/app/theprojectshop/Kassy/file/2531448132/issues/source?bid=3282987&fileBranchId=3283883#l7)

File: core/output/grunt.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
+var shim = require.once('../shim.js'),
+    server = require.once('http').createServer();
+    io = require.once('socket.io')(server),
```
Comment (mrkno) [inline core/output/grunt.js]: Codacy Issue found: [Expected an assignment or function call and instead saw an expression.](https://www.codacy.com/app/theprojectshop/Kassy/file/2531448095/issues/source?bid=3282987&fileBranchId=3283883#l3)

File: core/output/grunt.js [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+io.on('connection', function(s) {
+    console.debug('connection received');
+    if (socket !== null) {
+        throw 'existing connection already established';
+    }
+    socket = s;
+
+    socket.on('message', function(data) {
+        console.debug("received message: " + data);
```
Comment (mrkno) [inline core/output/grunt.js]: Codacy Issue found: [Strings must use singlequote.](https://www.codacy.com/app/theprojectshop/Kassy/file/2531448095/issues/source?bid=3282987&fileBranchId=3283883#l114)

File: core/output/grunt.js [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+        recMessage(data);
+    });
+
+    socket.on('error', function(error) {
+        console.error(error);
+    });
+
+    socket.on('disconnect', function() {
+        console.debug('disconnected');
+    })
```
Comment (mrkno) [inline core/output/grunt.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/theprojectshop/Kassy/file/2531448095/issues/source?bid=3282987&fileBranchId=3283883#l124)

File: core/output/grunt.js [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+            }
+            if (!data.content) {
+                data.content = '';
+            }
+
+            var event = shim.createEvent(data.thread_id, data.sender_id, data.sender_name, data.content);
+            return callback(api, event);
+        }
+        console.error('Received messsage with no content');
+    };
```
Comment (mrkno) [inline core/output/grunt.js]: Codacy Issue found: [Expected an assignment or function call and instead saw an expression.](https://www.codacy.com/app/theprojectshop/Kassy/file/2531448095/issues/source?bid=3282987&fileBranchId=3283883#l82)

File: core/output/grunt.js (hunk 4 above)
Comment (mrkno) [inline core/output/grunt.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/theprojectshop/Kassy/file/2531448095/issues/source?bid=3282987&fileBranchId=3283883#l114)

File: core/output/grunt.js (hunk 5 above)
Comment (mrkno) [inline core/output/grunt.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/theprojectshop/Kassy/file/2531448095/issues/source?bid=3282987&fileBranchId=3283883#l124)

**Comment by Awarua-** [comment]
Still to fix Watch functionality, currently it will try and launch multiple Kassy instances

File: test/acceptance/test_core_modules.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
+var assert = require('chai').assert,
+    c = require('../helpers/client.js');
+
+describe('Test core modules', function() {
+    this.timeout(5000);
+    before(function(done) {
+        client = new c(done);
```
Comment (mrkno) [inline test/acceptance/test_core_modules.js]: Codacy Issue found: [A constructor name should not start with a lowercase letter.](https://www.codacy.com/app/theprojectshop/Kassy/file/2532315321/issues/source?bid=3282987&fileBranchId=3283883#l7)

Comment (mrkno) [inline test/acceptance/test_core_modules.js]: Codacy Issue found: [Variable 'client' implicitly declared.](https://www.codacy.com/app/theprojectshop/Kassy/file/2532315321/issues/source?bid=3282987&fileBranchId=3283883#l7)

File: core/output/grunt.js (hunk 3 above)
Comment (mrkno) [inline core/output/grunt.js]: Codacy Issue found: [Expected an assignment or function call and instead saw an expression.](https://www.codacy.com/app/theprojectshop/Kassy/file/2532315284/issues/source?bid=3282987&fileBranchId=3283883#l3)

File: core/output/grunt.js (hunk 6 above)
Comment (mrkno) [inline core/output/grunt.js]: Codacy Issue found: [Expected an assignment or function call and instead saw an expression.](https://www.codacy.com/app/theprojectshop/Kassy/file/2532315284/issues/source?bid=3282987&fileBranchId=3283883#l82)

File: test/unit/test.js [hunk 8]
```diff
@@ -0,0 +1,10 @@
+var assert = require('chai').assert;
+
+describe('Array', function() {
+  describe('#indexOf()', function () {
+    it('should return -1 when the value is not present', function () {
+      assert.equal(-1, [1,2,3].indexOf(5));
+      assert.equal(-1, [1,2,3].indexOf(0));
```
Comment (mrkno) [inline test/unit/test.js]: Codacy Issue found: [A space is required after ','.](https://www.codacy.com/app/theprojectshop/Kassy/file/2532315323/issues/source?bid=3282987&fileBranchId=3283883#l7)

File: test/unit/test.js [hunk 9]
```diff
@@ -0,0 +1,10 @@
+var assert = require('chai').assert;
+
+describe('Array', function() {
+  describe('#indexOf()', function () {
+    it('should return -1 when the value is not present', function () {
+      assert.equal(-1, [1,2,3].indexOf(5));
```
Comment (mrkno) [inline test/unit/test.js]: Codacy Issue found: [A space is required after ','.](https://www.codacy.com/app/theprojectshop/Kassy/file/2532315323/issues/source?bid=3282987&fileBranchId=3283883#l6)

File: test/acceptance/test_core_modules.js (hunk 7 above)
Comment (mrkno) [inline test/acceptance/test_core_modules.js]: Codacy Issue found: [Variable 'client' implicitly declared.](https://www.codacy.com/app/theprojectshop/Kassy/file/2532328761/issues/source?bid=3282987&fileBranchId=3283883#l7)

Comment (mrkno) [inline test/acceptance/test_core_modules.js]: Codacy Issue found: [A constructor name should not start with a lowercase letter.](https://www.codacy.com/app/theprojectshop/Kassy/file/2532328761/issues/source?bid=3282987&fileBranchId=3283883#l7)

File: core/core_integrations/grunt.js [hunk 10]
```diff
@@ -0,0 +1,125 @@
+var shim = require.once('../shim.js'),
+    server = require.once('http').createServer();
+    io = require.once('socket.io')(server),
```
Comment (mrkno) [inline core/core_integrations/grunt.js]: Codacy Issue found: [Expected an assignment or function call and instead saw an expression.](https://www.codacy.com/app/theprojectshop/Kassy/file/2532328701/issues/source?bid=3282987&fileBranchId=3283883#l3)

File: core/core_integrations/grunt.js [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+            }
+            if (!data.content) {
+                data.content = '';
+            }
+
+            var event = shim.createEvent(data.thread_id, data.sender_id, data.sender_name, data.content);
+            return callback(api, event);
+        }
+        console.error('Received messsage with no content');
+    };
```
Comment (mrkno) [inline core/core_integrations/grunt.js]: Codacy Issue found: [Expected an assignment or function call and instead saw an expression.](https://www.codacy.com/app/theprojectshop/Kassy/file/2532328701/issues/source?bid=3282987&fileBranchId=3283883#l82)

File: test/unit/test.js (hunk 8 above)
Comment (mrkno) [inline test/unit/test.js]: Codacy Issue found: [A space is required after ','.](https://www.codacy.com/app/theprojectshop/Kassy/file/2532328763/issues/source?bid=3282987&fileBranchId=3283883#l7)

File: test/unit/test.js (hunk 9 above)
Comment (mrkno) [inline test/unit/test.js]: Codacy Issue found: [A space is required after ','.](https://www.codacy.com/app/theprojectshop/Kassy/file/2532328763/issues/source?bid=3282987&fileBranchId=3283883#l6)

File: core/core_integrations/grunt.js (hunk 10 above)
Comment (mrkno) [inline core/core_integrations/grunt.js]: Codacy Issue found: [Expected an assignment or function call and instead saw an expression.](https://www.codacy.com/app/theprojectshop/Kassy/file/2532331355/issues/source?bid=3282987&fileBranchId=3283883#l3)

File: core/core_integrations/grunt.js (hunk 11 above)
Comment (mrkno) [inline core/core_integrations/grunt.js]: Codacy Issue found: [Expected an assignment or function call and instead saw an expression.](https://www.codacy.com/app/theprojectshop/Kassy/file/2532331355/issues/source?bid=3282987&fileBranchId=3283883#l82)

Comment (mrkno) [inline core/core_integrations/grunt.js]: Codacy Issue found: [Expected an assignment or function call and instead saw an expression.](https://www.codacy.com/app/theprojectshop/Kassy/file/2532349301/issues/source?bid=3282987&fileBranchId=3283883#l82)

File: core/core_integrations/grunt.js (hunk 10 above)
Comment (mrkno) [inline core/core_integrations/grunt.js]: Codacy Issue found: [Expected an assignment or function call and instead saw an expression.](https://www.codacy.com/app/theprojectshop/Kassy/file/2532349301/issues/source?bid=3282987&fileBranchId=3283883#l3)

**Comment by Awarua-** [comment]
Watch has now been fixed

**Comment by Awarua-** [comment]
Adds #67 functionality

**Comment by mrkno** [comment]
Approved, ignoring codacy.

**Pull request merged**